src/sentry/snuba/metrics/extraction.py

```diff
@@ -359,24 +384,91 @@ def _transform_search_query(query: Sequence[QueryToken]) -> Sequence[QueryToken]
     return transformed_query
 
 
-def _parse_search_query(
-    query: Optional[str], removed_blacklisted: bool = False
+def parse_search_query(
+    query: Optional[str],
+    removed_blacklisted: bool = False,
+    force_transaction_event_type: bool = False,
```

Comment:
I don't see a reason to pass these flags to parse_search_query, this should be done by the callers.

Reply:
I was torn quite a bit around this. I also don't like the params but I was thinking that these two behaviors are reused quite often and could be seen as some sort of configuration to a parser of the query. I wanted to make the behavior opaque to avoid having callers deal with all the control flow logic of passing mutated tokens between function calls (which can be pretty easy to screw up since the order of transformations/validations is important).